lib/types.d.ts

```diff
@@ -0,0 +1,40 @@
+import {IsEqual, ValueOf} from 'type-fest';
+
+type MergeBy<TObject, TOtherObject> = Omit<TObject, keyof TOtherObject> & TOtherObject;
+
+type DeepRecord<TKey extends string | number | symbol, TValue> = {[key: string]: TValue | DeepRecord<TKey, TValue>};
+
+type TypeOptions = MergeBy<
+    {
+        keys: string;
+        collectionKeys: string;
+        values: Record<string, unknown>;
+    },
+    CustomTypeOptions
+>;
+
+interface CustomTypeOptions {}
+
+type Key = TypeOptions['keys'];
+type CollectionKeyBase = TypeOptions['collectionKeys'];
+type CollectionKey = `${CollectionKeyBase}${string}`;
+
+type OnyxKey = Key | CollectionKey;
+type KeyValueMapping = TypeOptions['values'];
+
+type GetOnyxValue<K extends Key | `${CollectionKeyBase}${string}`> = K extends Key
```

Comment:
This looks like a draft, but we should say TKey instead of K

Reply:
@neil-marcellini Yes, we are still working on this type.